lib/combine_pdf/decrypt.rb

```diff
@@ -143,12 +140,16 @@ def decrypt_AES(encrypted, encrypted_id, encrypted_generation, _encrypted_filter
       object_key << 'sAlT'.force_encoding(Encoding::ASCII_8BIT)
       key_length = object_key.length < 16 ? object_key.length : 16
 
-      cipher = OpenSSL::Cipher.new("aes-#{key_length << 3}-cbc")
-      cipher.decrypt
-      cipher.key = Digest::MD5.digest(object_key)[(0...key_length)]
-      cipher.iv = encrypted[0..15]
-      cipher.padding = 0
-      cipher.update(encrypted[16..-1]) + cipher.final
+      begin
+        cipher = OpenSSL::Cipher.new("aes-#{key_length << 3}-cbc")
+        cipher.decrypt
+        cipher.key = Digest::MD5.digest(object_key)[(0...key_length)]
+        cipher.iv = encrypted[0..15]
+        cipher.padding = 0
+        cipher.update(encrypted[16..-1]) + cipher.final
+      rescue Exception => e
```

Comment:
Rescuing all exceptions should be avoided. We should limit the exceptions we rescue to "expected" exceptions.

For example, when an server application is shut down, the Interrupt signal is raised using an "INT" signal which is also an exception in Ruby. That interrupt signal should be allowed to go up the stack to the server controller (which would probably contain a signal handler). By catching this signal here, we will prevent servers from performing a graceful shutdown, which is often used when horizontal scaling.